zcash_primitives/src/merkle_tree.rs

```diff
@@ -164,14 +164,8 @@ impl<Node: Hashable> CommitmentTree<Node> {
         //    - Empty leaves are used as needed.
         let leaf_root = Node::combine(
             0,
-            &match self.left {
-                Some(node) => node,
-                None => filler.next(0),
-            },
-            &match self.right {
-                Some(node) => node,
-                None => filler.next(0),
-            },
+            &self.left.unwrap_or_else(|| filler.next(0)),
+            &self.right.unwrap_or_else(|| filler.next(0)),
```

Comment:
I [checked](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=db5250068e149d9d2ed5fcfbf0e41b3c) that `filler.next(0)` inside `unwrap_or_else` closures would still be called in the expected order (left then right).